pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -0,0 +1,859 @@
+        :param offset: Will be a float valid between some valid range which depends on the step size
+        and step generator source. The instrument only will consider increments of the 10%
+        of the step size, other values will not have effect on the step generator offset.
+        """
+        stepgen_step_source_and_size = self.stepgen_step_source_and_size
+        stepgen_step_size = stepgen_step_source_and_size[1]
+        stepgen_step_source = stepgen_step_source_and_size[0]
+        max_offset = 5.0 * stepgen_step_size
+
+        if abs(offset) > max_offset:
+            offset_units = "V" if "VOLTAGE" in stepgen_step_source else "A"
+            raise ValueError(
+                f'Setting the step generator offset....\n'
+                f'Step Generator Offset cannot be greater than the maximum offset done the actual'
+                f' step size.\n'
+                f'Actual step size = " {stepgen_step_size} {offset_units}.\n'
+                f'Then maximun offset is {max_offset} {offset_units}.\n'
+                f'You wanted {offset} {offset_units}\n'
+            )
+
+        invert = False
+        if offset < 0:
+            invert = True
+
+        multiplier = abs(offset) / stepgen_step_size
+
+        self.stepgen_step_size_multiplier = multiplier
+        if not invert == self.stepgen_invert:
+            self.stepgen_invert = invert
+
+    ################################################################################
+    # MISCELLANEOUS COMMANDS AND QUERIES
+    # The miscellaneous commands and queries group contains queries for the status
+    # of the output connectors, and the measurement code, as well as commands to set
+    # the measurement mode and save and recall sets of front-panel settings.
+    ################################################################################
+
+    measure_mode = Instrument.control(
+        "MEAsure?", "MEAsure %s",
+        """Control the measurement mode (string).""",
+        validator=strict_discrete_set,
+        values=VALID_MEASUREMENT_MODES,
+        get_process=lambda r:
+        r.replace("MEASURE ", "")
+    )
+
+    #########################################################################################
+    # WAVEFORM TRANSFER COMMANDS AND QUERIES
+    # The Waveform Transfer group allows curve or preamble data (or both) to be stored in,
+    # or recalled from, mass storage.
+    # There is also a command to set the number of curve data points stored
+    # and a related query to determine the length of a previously defined waveform.
+    #########################################################################################
+
+    waveform_preamble = Instrument.control(
+        "WFMpre?", "WFMpre %s",
+        """Control the waveform preamble data for the currently displayed waveform (string).""",
+        # Response syntax
+        get_process=lambda r:
+        r
+    )
+
+    def curve(self, bytes_count):
```

Comment:
If it is a method, it should have an "action" name, i.e. a verb, like "fetch_curve".

Reply:
Right. Thanks